homeassistant/core.py

```diff
@@ -2136,6 +2171,12 @@ def async_set(
             context=context,
             time_fired=timestamp,
         )
+        self._bus._async_fire(  # pylint: disable=protected-access
+            EVENT_STATE_REPORTED,
```

Comment:
I'm wondering if we could save this call by adding an alias lookup for the event type in `_async_fire`. We know that all state changed events matches state reported events exactly. So we could alias the state changed event with the state reported event. Not the other way around though.

Reply:
That's a really good idea, thanks @MartinHjelmare 👍